File: Year 1/Semester 1/Java/Lessons/classes/outer_class.java
package classes;

    public class outer_class {
        private String outerField = "Outer field";
    
        // Inner class
        public class InnerClass {
            private String innerField = "Inner field";
    
            public void display() {
                // Accessing outer class field
                System.out.println("Outer field: " + outerField);
                // Accessing inner class field
                System.out.println("Inner field: " + innerField);
            }
        }
    
        public void outerMethod() {
            // Creating an instance of the inner class
            InnerClass innerObject = new InnerClass();
            innerObject.display();
        }
    
        public static void main(String[] args) {
            // Creating an instance of the outer class
            outer_class outerObject = new outer_class();
            // Calling the outer class method, which in turn uses the inner class
            outerObject.outerMethod();
        }
    
    
}

File: Year 1/Semester 1/C/Lessons/classes/outer_class.c
#include <stdio.h>
#include <string.h>

// Outer class equivalent (struct)
typedef struct {
    char outerField[50];
} OuterClass;

// Inner class equivalent (struct)
typedef struct {
    char innerField[50];
    OuterClass *outer;  // Reference to the outer class
} InnerClass;

// Method to display the outer and inner fields
void display(InnerClass *inner) {
    printf("Outer field: %s\n", inner->outer->outerField);
    printf("Inner field: %s\n", inner->innerField);
}

// Method equivalent to outerMethod()
void outerMethod(OuterClass *outer) {
    InnerClass inner;
    inner.outer = outer;  // Link to the outer class
    strcpy(inner.innerField, "Inner field");

    display(&inner);  // Calling inner class's display method
}

int main() {
    // Creating an instance of the outer class
    OuterClass outer;
    strcpy(outer.outerField, "Outer field");

    // Calling the outer method, which uses the inner class
    outerMethod(&outer);

    return 0;
}

